### PR TITLE
Updated to the new image server.

### PR DESCRIPTION
**Pull request by StinGer-ShoGuN**
Actually, this is more of a 'hack' than a real fix. I'm saying this because maybe, the way images are handled should be reviewed; I don't know the sources enough to tell.

It might be incomplete. I'll update it if necessary.

See https://developers.eveonline.com/blog/article/from-image-server-to-a-whole-new-image-service-1 for details.

### CODE REVIEW

File: common/includes/class.thumb.php [hunk 1]
```diff
@@ -273,15 +273,24 @@ function genNPC()
...
         if($this->encoding == 'jpeg') $ext = 'jpg';
         else $ext = 'png';
 
-        if($type != 'Character' && $type != 'Corporation' && $type != 'Alliance') return false;
-        $url = IMG_SERVER."/".$type."/".$this->id."_".$size.".".$ext;
+        if($type != 'characters' && $type != 'corporations' && $type != 'alliances') return false;
+        $url = IMG_SERVER."/".$type."/".$this->id;
+        if ($type == 'characters')
+        {
+            $url .= "portrait?size=${size}";
```
Comment (Salvoxia) [inline common/includes/class.thumb.php]: Needs to start with a slash to build the URL correctly:
```suggestion
            $url .= "/portrait?size=${size}";
```

File: common/includes/class.thumb.php [hunk 2]
```diff
@@ -273,15 +273,24 @@ function genNPC()
...
+        }
+        else
+        {
+            $url .= "logo?size=${size}";
```
Comment (Salvoxia) [inline common/includes/class.thumb.php]: Needs to start with a slash to build the URL correctly:
```suggestion
            $url .= "/logo?size=${size}";
```

**Review by Salvoxia — CHANGES_REQUESTED** [review]
I made two change suggestions so that the URL is constructed correctly for character portraits and alliance/corporation logos.
Also, the following change is required for corp logos to be fetched:
[Line 195 of common/includes/class.thumb.php](https://github.com/evekb/evedev-kb/pull/112/files#diff-06710fbcf49b8258e6372cd529bd2f16R195) needs to be 
`$img = $this->fetchImage('corp', 128);`

Could you please incorporate these changes, then I'll merge the changes.

**Comment by StinGer-ShoGuN** [comment]
Nice catch !

Where are thumbs used in the KB ? Because I didn't see any errors (404) on mine...

**Comment by Salvoxia** [comment]
Thank you very much for your contribution!
Do you know what's up with some of the NPC logos on the new image server?
The Faction "Unknown" does not have a logo, even though most of the other factions do:
https://images.evetech.net/corporations/500021/logo
vs.
https://images.evetech.net/corporations/500026/logo

Same for Drifters:
https://images.evetech.net/corporations/1000274/logo

**Comment by StinGer-ShoGuN** [comment]
I have no idea... Digging a bit I realised the server returns a 404 on those with a x-cache error "Error from cloudfront". Maybe it's an issue from CCP CDN.

I've posted [a message](https://forums.eveonline.com/t/new-image-server-again/202922) on EVE forums. Maybe we'll have an answer.

**Comment by soratidus999** [comment]
There are many issues with the new image server currently.  Many alliance logos have been available/unavailable at various times over the last several weeks

**Pull request merged**